avm/utilities/pipelines/platform/Sync-AvmModulesList.ps1

```diff
@@ -0,0 +1,64 @@
+<#
+.SYNOPSIS
+Updating the module names list in the issue template
+
+.DESCRIPTION
+CSV data for moules and pattern is loaded and overwrites the list in the issue template. The changes are then commited to the repository.
+
+.PARAMETER Repo
+Repository name according to GitHub (owner/name)
+
+.PARAMETER RepoRoot
+Optional. Path to the root of the repository.
+
+.EXAMPLE
+Sync-AvmModulesList -Repo 'Azure/bicep-registry-modules'
+
+.NOTES
+Will be triggered by the workflow avm.platform.sync-avm-modules-list.yml
+#>
+function Sync-AvmModulesList {
+  param (
+    [Parameter(Mandatory = $true)]
+    [string] $Repo,
+
+    [Parameter(Mandatory = $false)]
+    [string] $RepoRoot = (Get-Item -Path $PSScriptRoot).parent.parent.parent.parent.FullName
+  )
+
+  $workflowFilePath = Join-Path $RepoRoot '.github' 'ISSUE_TEMPLATE' 'avm_module_issue.yml'
+
+  # get CSV data
+  $modules = Get-AvmCsvData -ModuleIndex "Bicep-Resource" | Select-Object -Property "ModuleName"
+  $patterns = Get-AvmCsvData -ModuleIndex "Bicep-Pattern" | Select-Object -Property "ModuleName"
+
+  # build new strings
+  $prefix = '        - "'
+  $postfix = '"'
+  $newModuleLines = $modules | ForEach-Object { $prefix + $_.ModuleName + $postfix }
+  $newPatternLines = $patterns | ForEach-Object { $prefix + $_.ModuleName + $postfix }
+
+  # parse workflow file
+  $workflowFileLines = Get-Content $workflowFilePath
+  $startIndex = 0
+  $endIndex = 0
+
+  for ($lineNumber = 0; $lineNumber -lt $workflowFileLines.Count; $lineNumber++) {
+    if ($startIndex -gt 0 -and (-not $workflowFileLines[$lineNumber].StartsWith('        - "avm/'))) {
```

Comment:
The spaces here are a pretty big gamble. I'd highly recommend to refactoring this to a regex .e.g via
`if ($startIndex -gt 0 -and (-not $workflowFileLines[$lineNumber] -match '\s*- "avm\/')) {` 

And the same could be set for other lines. Also, instead of guessing the indent, you could have regex figure it out for you and add new items with the same count. However - as this is a first version I don't want to push you to do that and could offer to give it a short once merged. If you agree, please reply and we can resolve this comment.

Reply:
I introduced trim, although it is not exactly what you are suggesting. Let's move it to the next version